sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableCatalog.java

```diff
@@ -162,6 +162,29 @@ Table alterTable(
    */
   boolean dropTable(Identifier ident);
 
+  /**
+   * Drop a table in the catalog with an option to purge.
+   * <p>
+   * If the catalog supports views and contains a view for the identifier and not a table, this
+   * must not drop the view and must return false.
+   * <p>
+   * If the purge option is set to false, the default implementation falls back to
+   * {@link #dropTable(Identifier)} dropTable}. Otherwise, it throws
+   * {@link UnsupportedOperationException}.
```

Comment:
we should mention that, people need to overwrite this method to support the purge option.